### ISSUE TITLE
Build fails during libgcrypt on Arch Linux

### ISSUE THREAD

**Issue by flawedworld** [osresearch/heads#585]
Attempting to compile heads results in libgcrypt failing to build, it is present and have tested on Debian with same steps and is an issue with the way libgcrypt is installed on Arch Linux. 

Terminal output: https://pastebin.com/9GC168wB

**Comment by tlaurion** [comment]
Please do:
```
make BOARD=t430 real.clean
rm -rf build ./build/*
make BOARD=t430 V=1
```
And post back build logs. Never build on arch.

**Comment by flawedworld** [comment]
@tlaurion 

Done so exactly as you asked, here's the logs. From newest to oldest.

gpg2.log https://pastebin.com/GiS0U7pj

gpg2.configure.log https://pastebin.com/WdrbPzLm

libgcrypt.log https://pastebin.com/biycZafz

libgcrypt.configure.log https://pastebin.com/bjSRNjz1

**Comment by tlaurion** [comment]
Gpg2 logs shows missing glibc dependencies from the errors .

Followed building docs on heads-wiki?
Have you installed arch equivalents of fedora/debian dependencies?



On June 20, 2019 2:05:49 PM UTC, "Gabe Gałązka" <notifications@github.com> wrote:
>@tlaurion 
>
>Done so exactly as you asked, here's the logs. From newest to oldest.
>
>gpg2.log https://pastebin.com/GiS0U7pj
>
>gpg2.configure.log https://pastebin.com/WdrbPzLm
>
>libgcrypt.log https://pastebin.com/biycZafz
>
>libgcrypt.configure.log https://pastebin.com/bjSRNjz1
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/osresearch/heads/issues/585#issuecomment-504039221

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by tlaurion** [comment]
The error is

/home/user/Project/heads_bios/seb/heads/crossgcc/x86_64-linux-musl/lib/gcc/x86_64-linux-musl/5.3.0/../../../../x86_64-linux-musl/bin/ld: warning: libc.so.6, needed by /home/user/Project/heads_bios/seb/heads/install/lib/libgcrypt.so, not found (try using -rpath or -rpath-link)

/home/user/Project/heads_bios/seb/heads/install/lib/libgcrypt.so: undefined reference to `free@GLIBC_2.2.5'



On June 20, 2019 2:05:49 PM UTC, "Gabe Gałązka" <notifications@github.com> wrote:
>@tlaurion 
>
>Done so exactly as you asked, here's the logs. From newest to oldest.
>
>gpg2.log https://pastebin.com/GiS0U7pj
>
>gpg2.configure.log https://pastebin.com/WdrbPzLm
>
>libgcrypt.log https://pastebin.com/biycZafz
>
>libgcrypt.configure.log https://pastebin.com/bjSRNjz1
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/osresearch/heads/issues/585#issuecomment-504039221

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by tlaurion** [comment]
@flawedworld any update?

**Comment by flawedworld** [comment]
@tlaurion  So first off, sorry for the lateness, Kinda forgot about this since I'm waiting for this to be brought up to a newer coreboot release. Anyways.

base-devel, gcc-ada, cpio, cmake, iasl, nasm, ccache, perl-digest-sha1 were installed on a fresh Arch install. uuid was sourced from the AUR. All other packages mentioned in the docs were present in either base or base-devel. I no longer run Arch daily FYI.

I was able to compile x230-flash perfectly. Compiling x230 gives me an error during libgpg-error.  https://pastebin.com/u0F6KXwm    https://pastebin.com/i38WXwwr

System was freshly installed, with literally just the above packages. All fully updated.

A potential solution would be the option to compile heads within a docker image, which I might play around with in my downtime. If you need any other logs please let me know.

**Comment by flawedworld** [comment]
Closed, see issue #668
